static/index.js

```diff
@@ -0,0 +1,82 @@
+let reviews = [];
+
+let sum = 0;
+
+if(localStorage.getItem("reviews") !== null) reviews = JSON.parse(localStorage.getItem("reviews"));
+
+
+
+const updateAverage = () =>{
+    document.querySelector("#averageRating").innerText = `${Math.round(sum / reviews.length * 10) / 10}/5`;
+}
+
+const createItem = (item) => {
+    let reviewItem = document.createElement("div");
+    let reviewText = document.createElement("p");
+    reviewText.innerText = item.review;
+    
+    let ratingText = document.createElement("p");
+    ratingText.innerText = `${item.rating} / 5`;
+
+    let deleteButton = document.createElement("button");
+    deleteButton.innerText = "Delete";
+    deleteButton.id = `item-${item.id}`;
+    
+    updateAverage();
+
+    deleteButton.addEventListener("click", (e)=>{
+        for(let rev in reviews){
+            console.log(`${e.target.id.split('-')[1] } VS..... ${reviews[rev].id}`)
+
+            if(e.target.id.split('-')[1] == reviews[rev].id) {
+                sum -= reviews[rev].rating;
+                console.log("brisam", reviews[rev]);
+                reviews.splice(rev, 1);
+                break;
+            }
+            
+        }
+    
+        localStorage.setItem("reviews", JSON.stringify(reviews));
```

Comment:
Imagine that in the next homework, the assignment is to save the reviews in sessionStorage instead of in localStorage. You would have to do this in 2 places. In a bigger application, you would have to do this in 10+ places. The probability is high that you would forget or mess up doing it in one place and introduce a bug.

That is why it is a good idea to create a function called `saveReviews` or similar, that does this for you. That way, if you want to switch from localStorage to sessionStorage, you only need to do this in one place.
This is a very important benefit of abstraction.

Reply:
Fixed in [Improvements](https://github.com/infinum-academy/js-react-academy-2024-filiptodosov/pull/1/commits/79575e604ff61d5206a96a781ad054151af9c333)